dmscripts/export_framework_suppliers.py

```diff
@@ -0,0 +1,390 @@
+# -*- coding: utf-8 -*-
+import jsonschema
+import os
+import sys
+if sys.version_info[0] < 3:
+    import unicodecsv as csv
+else:
+    import csv
+
+from multiprocessing.pool import ThreadPool
+
+
+DRAFT_STATUSES = [
+    'completed',
+    'failed',
+    'draft',
+]
+LOTS = [
+    'digital-outcomes', 'digital-specialists',
+    'user-research-studios', 'user-research-participants',
+]
+
+
+def get_validation_errors(candidate, schema):
+        validator = jsonschema.Draft4Validator(schema)
+        errors = validator.iter_errors(candidate)
+        error_keys = [error.path[0] for error in errors]
+        return error_keys
+
+
+def find_suppliers(client, framework_slug):
+    suppliers = client.get_interested_suppliers(framework_slug)['interestedSuppliers']
+    return ({'supplier_id': supplier_id} for supplier_id in suppliers)
+
+
+def add_supplier_info(client):
+    def inner(record):
+        supplier = client.get_supplier(record['supplier_id'])
+
+        return dict(record, supplier=supplier['suppliers'])
+
+    return inner
+
+
+def add_draft_services(client, framework_slug, lot=None, status=None):
+    def inner(record):
+        drafts = client.find_draft_services(record["supplier_id"], framework=framework_slug)
+        drafts = drafts["services"]
+
+        drafts = [
+            draft for draft in drafts
+            if (not lot or draft["lotSlug"] == lot) and (not status or draft["status"] == status)
+        ]
+
+        return dict(record,
+                    services=drafts)
+
+    return inner
+
+
+def count_field_in_record(field_id, field_label, record):
+    return sum(1
+               for service in record["services"]
+               if field_label in service.get(field_id, []))
+
+
+def make_field_title(field_id, field_label):
+    return "{} {}".format(field_id, field_label)
+
+
+def make_fields_from_content_question(question, record):
```

Comment:
I think actually `fields_from_content_question` would be a better name for this - `make_` implies almost that it has a side effect. In fact in the past, often when writing a generator I've been explicit that it's a generator - `fields_from_content_question_gen` or something...